DuckDuckGo/NetworkProtectionStatusView.swift

```diff
@@ -359,11 +330,26 @@ struct NetworkProtectionStatusView: View {
     @ViewBuilder
     private func geoswitchingTipView() -> some View {
         if statusModel.canShowTips {
-
-            TipView(geoswitchingTip)
+            TipView(tipsModel.geoswitchingTip)
                 .removeGroupedListStyleInsets()
                 .tipCornerRadius(0)
                 .tipBackground(Color(designSystemColor: .surface))
+                .onAppear {
+                    tipsModel.handleGeoswitchingTipShown()
+                }
+                .task {
+                    var previousStatus = tipsModel.geoswitchingTip.status
+
+                    for await status in tipsModel.geoswitchingTip.statusUpdates {
+                        if case .invalidated(let reason) = status {
+                            if case .available = previousStatus {
```

Comment:
This is a bit nasty.  I copied this logic from macOS.

Tips don't have a simple action hook for when they're dismissed through the "X" button, unless we reimplement them from scratch.

Since I didn't want to recreate the standard tip UI, I am observing the tip status and when I detect a change from `available` (should be shown to the user) to `invalidated` (either closed or actioned) I'm firing a call to handle the event.  This is mostly used for pixel-firing within `tipsModel`.